tests/runtime-tests/src/lib.rs

```diff
@@ -220,52 +151,54 @@ fn run_test(test_path: &Path, mut spin: Spin, on_error: OnTestError) {
 /// Copies the test dir's manifest file into the temporary directory
 ///
 /// Replaces template variables in the manifest file with components from the components path.
-fn copy_manifest(test_dir: &Path, temp: &temp_dir::TempDir) -> anyhow::Result<()> {
+fn copy_manifest(
+    test_dir: &Path,
+    temp: &temp_dir::TempDir,
+    services: &Services,
+) -> anyhow::Result<()> {
     let manifest_path = test_dir.join("spin.toml");
-    let manifest = std::fs::read_to_string(&manifest_path).with_context(|| {
+    let mut manifest = std::fs::read_to_string(manifest_path).with_context(|| {
         format!(
             "no spin.toml manifest found in test directory {}",
             test_dir.display()
         )
     })?;
-    let mut table = manifest
-        .parse::<toml::Table>()
-        .context("could not parse the manifest as toml")?;
-    for (_, component) in table["component"].as_table_mut().with_context(|| {
-        format!(
-            "malformed manifest '{}' does not contain 'component' array",
-            manifest_path.display(),
-        )
-    })? {
-        let source = component.get_mut("source").with_context(|| {
-            format!(
-                "malformed manifest '{}' does not contain 'source' string key in component",
-                manifest_path.display()
-            )
-        })?;
-        let source_str = source.as_str().with_context(|| {
-            format!(
-                "malformed manifest '{}' contains 'source' key that isn't a string in component",
-                manifest_path.display()
-            )
+    let regex = regex::Regex::new(r"\{\{(.*)\}\}").unwrap();
+    while let Some(captures) = regex.captures(&manifest) {
+        let (Some(full), Some(capture)) = (captures.get(0), captures.get(1)) else {
+            continue;
+        };
+        let template = capture.as_str();
+        let (template_key, template_value) = template.split_once('=').with_context(|| {
+            format!("invalid template '{template}'(template should be in the form $KEY=$VALUE)")
```

Comment:
Am I reading this right that it would preclude the use of spin variables in a test manifest?

Reply:
I think the point you brought up above about using a different template delimiter applies here. I just completely forgot that `{{}}` already has semantic meaning. Perhaps `${{}}` would work?

Reply:
`${{}}` might be confusing as it could be read as "`$` + <variable>".